several `Status` in OPs are not checked

### Summary

The compilation warnings are raised.

They have the risk of segment fault due to not being checked.

### DeePMD-kit Version

devel

### TensorFlow Version

2.12.0

### Python Version, CUDA Version, GCC Version, LAMMPS Version, etc

_No response_

### Details


```
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc: In instantiation of 'int _norm_copy_coord_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, FPTYPE*&, int*&, int*&, int&, int&, const FPTYPE*, const FPTYPE*, const int*, const int&, const int&, const float&) [with FPTYPE = double]':
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1644:41:   required from 'void _prepare_coord_nlist_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, const FPTYPE**, FPTYPE*&, const int**, int*&, int*&, deepmd::InputNlist&, int*&, int*&, int**&, int*&, int*&, int&, int&, int&, int&, const FPTYPE*, const int*, int, const int&, const int&, const float&, const int&, const int&) [with FPTYPE = double]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1182:41:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::_Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = double]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1005:59:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = double]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1003:8:   required from here
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1481:3: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1481 |   context->allocate_temp(DataTypeToEnum<FPTYPE>::value, FPTYPE_shape,
      |   ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1499:3: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1499 |   context->allocate_temp(DataTypeToEnum<FPTYPE>::value, double_shape,
      |   ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1523:5: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1523 |     context->allocate_temp(DataTypeToEnum<FPTYPE>::value, cpy_shape,
      |     ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc: In instantiation of 'int _build_nlist_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, int*&, int*&, int**&, int*&, int&, int&, const FPTYPE*, const int&, const int&, const int&, const float&) [with FPTYPE = double]':
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1654:25:   required from 'void _prepare_coord_nlist_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, const FPTYPE**, FPTYPE*&, const int**, int*&, int*&, deepmd::InputNlist&, int*&, int*&, int**&, int*&, int*&, int&, int&, int&, int&, const FPTYPE*, const int*, int, const int&, const int&, const float&, const int&, const int&) [with FPTYPE = double]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1182:41:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::_Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = double]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1005:59:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = double]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1003:8:   required from here
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1563:3: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1563 |   context->allocate_temp(DT_INT32, nlist_shape, tensor_list);
      |   ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1574:5: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1574 |     context->allocate_temp(DT_INT32, jlist_shape, tensor_list + 1);
      |     ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc: In instantiation of 'int _norm_copy_coord_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, FPTYPE*&, int*&, int*&, int&, int&, const FPTYPE*, const FPTYPE*, const int*, const int&, const int&, const float&) [with FPTYPE = float]':
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1644:41:   required from 'void _prepare_coord_nlist_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, const FPTYPE**, FPTYPE*&, const int**, int*&, int*&, deepmd::InputNlist&, int*&, int*&, int**&, int*&, int*&, int&, int&, int&, int&, const FPTYPE*, const int*, int, const int&, const int&, const float&, const int&, const int&) [with FPTYPE = float]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1182:41:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::_Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = float]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1005:59:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = float]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1003:8:   required from here
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1481:3: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1481 |   context->allocate_temp(DataTypeToEnum<FPTYPE>::value, FPTYPE_shape,
      |   ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1499:3: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1499 |   context->allocate_temp(DataTypeToEnum<FPTYPE>::value, double_shape,
      |   ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1523:5: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1523 |     context->allocate_temp(DataTypeToEnum<FPTYPE>::value, cpy_shape,
      |     ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc: In instantiation of 'int _build_nlist_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, int*&, int*&, int**&, int*&, int&, int&, const FPTYPE*, const int&, const int&, const int&, const float&) [with FPTYPE = float]':
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1654:25:   required from 'void _prepare_coord_nlist_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, const FPTYPE**, FPTYPE*&, const int**, int*&, int*&, deepmd::InputNlist&, int*&, int*&, int**&, int*&, int*&, int&, int&, int&, int&, const FPTYPE*, const int*, int, const int&, const int&, const float&, const int&, const int&) [with FPTYPE = float]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1182:41:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::_Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = float]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1005:59:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = float]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1003:8:   required from here
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1563:3: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1563 |   context->allocate_temp(DT_INT32, nlist_shape, tensor_list);
      |   ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1574:5: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1574 |     context->allocate_temp(DT_INT32, jlist_shape, tensor_list + 1);
      |     ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
```

--- a/source/op/prod_env_mat_multi_device.cc
+++ b/source/op/prod_env_mat_multi_device.cc
@@ -1501,8 +1501,11 @@ static int _norm_copy_coord_gpu(OpKernelContext* context,
   // Tensor FPTYPE_temp;
   TensorShape FPTYPE_shape;
   FPTYPE_shape.AddDim(nall * 3);
-  context->allocate_temp(DataTypeToEnum<FPTYPE>::value, FPTYPE_shape,
-                         tensor_list);
+  tensorflow::Status status = context->allocate_temp(
+      DataTypeToEnum<FPTYPE>::value, FPTYPE_shape, tensor_list);
+  if (!status.ok()) {
+    return false;
+  }
   FPTYPE* tmp_coord = (*tensor_list).flat<FPTYPE>().data();
   DPErrcheck(cudaMemcpy(tmp_coord, coord, sizeof(FPTYPE) * nall * 3,
                         cudaMemcpyDeviceToDevice));
@@ -1519,8 +1522,11 @@ static int _norm_copy_coord_gpu(OpKernelContext* context,
   // Tensor double_temp;
   TensorShape double_shape;
   double_shape.AddDim(18);
-  context->allocate_temp(DataTypeToEnum<FPTYPE>::value, double_shape,
-                         tensor_list + 1);
+  tensorflow::Status status = context->allocate_temp(
+      DataTypeToEnum<FPTYPE>::value, double_shape, tensor_list + 1);
+  if (!status.ok()) {
+    return false;
+  }
   // Tensor int_temp;
   TensorShape int_shape;
   int_shape.AddDim(23 + nloc * 3 + loc_cellnum + total_cellnum * 3 +
@@ -1583,7 +1589,11 @@ static int _build_nlist_gpu(OpKernelContext* context,
   // Tensor nlist_temp;
   TensorShape nlist_shape;
   nlist_shape.AddDim(nloc * 2);
-  context->allocate_temp(DT_INT32, nlist_shape, tensor_list);
+  tensorflow::Status status =
+      context->allocate_temp(DT_INT32, nlist_shape, tensor_list);
+  if (!status.ok()) {
+    return false;
+  }
   ilist = (*tensor_list).flat<int>().data();
   numneigh = ilist + nloc;
   // Tensor jlist_temp;
@@ -1594,7 +1604,11 @@ static int _build_nlist_gpu(OpKernelContext* context,
   for (tt = 0; tt < max_nnei_trial; ++tt) {
     TensorShape jlist_shape;
     jlist_shape.AddDim(3 * int_64(nloc) * mem_nnei);
-    context->allocate_temp(DT_INT32, jlist_shape, tensor_list + 1);
+    tensorflow::Status status =
+        context->allocate_temp(DT_INT32, jlist_shape, tensor_list + 1);
+    if (!status.ok()) {
+      return false;
+    }
     jlist = (*(tensor_list + 1)).flat<int>().data();
     ind_data = jlist + nloc * mem_nnei;
     for (int_64 ii = 0; ii < nloc; ++ii) {
@@ -1742,8 +1756,11 @@ static int _norm_copy_coord_gpu_rocm(OpKernelContext* context,
   // Tensor double_temp;
   TensorShape double_shape;
   double_shape.AddDim(18);
-  context->allocate_temp(DataTypeToEnum<FPTYPE>::value, double_shape,
-                         tensor_list + 1);
+  tensorflow::Status status = context->allocate_temp(
+      DataTypeToEnum<FPTYPE>::value, double_shape, tensor_list + 1);
+  if (!status.ok()) {
+    return false;
+  }
   // Tensor int_temp;
   TensorShape int_shape;
   int_shape.AddDim(23 + nloc * 3 + loc_cellnum + total_cellnum * 3 +
@@ -1806,7 +1823,11 @@ static int _build_nlist_gpu_rocm(OpKernelContext* context,
   // Tensor nlist_temp;
   TensorShape nlist_shape;
   nlist_shape.AddDim(nloc * 2);
-  context->allocate_temp(DT_INT32, nlist_shape, tensor_list);
+  tensorflow::Status status =
+      context->allocate_temp(DT_INT32, nlist_shape, tensor_list);
+  if (!status.ok()) {
+    return false;
+  }
   ilist = (*tensor_list).flat<int>().data();
   numneigh = ilist + nloc;
   // Tensor jlist_temp;
@@ -1817,7 +1838,11 @@ static int _build_nlist_gpu_rocm(OpKernelContext* context,
   for (tt = 0; tt < max_nnei_trial; ++tt) {
     TensorShape jlist_shape;
     jlist_shape.AddDim(3 * int_64(nloc) * mem_nnei);
-    context->allocate_temp(DT_INT32, jlist_shape, tensor_list + 1);
+    tensorflow::Status status =
+        context->allocate_temp(DT_INT32, jlist_shape, tensor_list + 1);
+    if (!status.ok()) {
+      return false;
+    }
     jlist = (*(tensor_list + 1)).flat<int>().data();
     ind_data = jlist + nloc * mem_nnei;
     for (int_64 ii = 0; ii < nloc; ++ii) {

--- a/source/op/prod_env_mat_multi_device.cc
+++ b/source/op/prod_env_mat_multi_device.cc
@@ -1522,8 +1522,8 @@ static int _norm_copy_coord_gpu(OpKernelContext* context,
   // Tensor double_temp;
   TensorShape double_shape;
   double_shape.AddDim(18);
-  tensorflow::Status status = context->allocate_temp(
-      DataTypeToEnum<FPTYPE>::value, double_shape, tensor_list + 1);
+  status = context->allocate_temp(DataTypeToEnum<FPTYPE>::value, double_shape,
+                                  tensor_list + 1);
   if (!status.ok()) {
     return false;
   }
@@ -1604,8 +1604,7 @@ static int _build_nlist_gpu(OpKernelContext* context,
   for (tt = 0; tt < max_nnei_trial; ++tt) {
     TensorShape jlist_shape;
     jlist_shape.AddDim(3 * int_64(nloc) * mem_nnei);
-    tensorflow::Status status =
-        context->allocate_temp(DT_INT32, jlist_shape, tensor_list + 1);
+    status = context->allocate_temp(DT_INT32, jlist_shape, tensor_list + 1);
     if (!status.ok()) {
       return false;
     }
@@ -1838,8 +1837,7 @@ static int _build_nlist_gpu_rocm(OpKernelContext* context,
   for (tt = 0; tt < max_nnei_trial; ++tt) {
     TensorShape jlist_shape;
     jlist_shape.AddDim(3 * int_64(nloc) * mem_nnei);
-    tensorflow::Status status =
-        context->allocate_temp(DT_INT32, jlist_shape, tensor_list + 1);
+    status = context->allocate_temp(DT_INT32, jlist_shape, tensor_list + 1);
     if (!status.ok()) {
       return false;
     }

